test/multi_target/multitarget_test.cpp

```diff
@@ -337,82 +312,55 @@ TEST_CASE(multitarget_compile_nested_if_then_else)
-        auto run_ins            = run_on_target_mod->add_instruction(
-            migraphx::make_op("run_on_target", {{"target_id", tid}}), inputs, {test_mod});
-        auto run_ins_0 = run_on_target_mod->add_instruction(
-            migraphx::make_op("get_tuple_elem", {{"index", 0}}), run_ins);
-        run_on_target_mod->add_return({run_ins_0});
-        return run_on_target_mod;
+        tass.insert(tass.begin(), std::make_pair(ins1, tid));
+        tass.insert(tass.begin(), std::make_pair(ins2, tid));
+        tass.insert(tass.begin(), std::make_pair(ins3, tid));
+        return test_mod;
     };
 
-    // create nested module with multiple targets.
-    // then_mod has one instruction that runs a module on "ref" and another instruction that
-    // creates nested modules using "If" that runs on "cpu" and "gpu"
-    auto* ref_mod = p.create_module("ref_mod");
-    auto ref_x    = ref_mod->add_parameter("ref_x", ds);
-    auto ref_y    = ref_mod->add_parameter("ref_y", ds);
-    auto ref_add  = ref_mod->add_instruction(migraphx::make_op("add"), ref_x, ref_y);
-    ref_mod->add_return({ref_add});
-
     auto* then_mod        = p.create_module("then_mod");
     auto then_mod_cond    = then_mod->add_parameter("then_mod_cond", cond_s);
     auto then_mod_param_0 = then_mod->add_parameter("then_mod_param_0", ds);
     auto then_mod_param_1 = then_mod->add_parameter("then_mod_param_1", ds);
     auto then_mod_param_2 = then_mod->add_parameter("then_mod_param_2", ds);
     auto then_mod_ref_ins =
-        then_mod->add_instruction(migraphx::make_op("run_on_target", {{"target_id", 3}}),
-                                  {then_mod_param_0, then_mod_param_1},
-                                  {ref_mod});
-    auto then_mod_ref_ins_0 = then_mod->add_instruction(
-        migraphx::make_op("get_tuple_elem", {{"index", 0}}), then_mod_ref_ins);
-    auto then_mod_if = then_mod->add_instruction(
-        migraphx::make_op("if"),
-        {then_mod_cond,
-         then_mod_param_0,
-         then_mod_param_1,
-         then_mod_param_2,
-         then_mod_ref_ins_0,
-         then_mod_param_1,
-         then_mod_param_2},
-        {create_test_module(p, {then_mod_param_0, then_mod_param_1, then_mod_param_2}, 1),
-         create_test_module(p, {then_mod_ref_ins_0, then_mod_param_1, then_mod_param_2}, 0)});
+        then_mod->add_instruction(migraphx::make_op("add"), then_mod_param_0, then_mod_param_1);
+    tass.insert(tass.begin(), std::make_pair(then_mod_ref_ins, 3));
+    auto then_mod_if =
+        then_mod->add_instruction(migraphx::make_op("if"),
+                                  {then_mod_cond,
+                                   then_mod_param_0,
+                                   then_mod_param_1,
+                                   then_mod_param_2,
+                                   then_mod_ref_ins,
+                                   then_mod_param_1,
+                                   then_mod_param_2},
```

Comment:
The `create_test_module` only creates 3 parameters, so this seems wrong to pass 6 parameters.

Reply:
It creates 3 parameters but for both then and else modules have different parameter names and therefore i need to pass 6 parameters.